server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -946,7 +946,8 @@ static Settings aggregateIndexSettings(
         if (INDEX_NUMBER_OF_SHARDS_SETTING.exists(indexSettingsBuilder) == false) {
             indexSettingsBuilder.put(SETTING_NUMBER_OF_SHARDS, INDEX_NUMBER_OF_SHARDS_SETTING.get(settings));
         }
-        if (INDEX_NUMBER_OF_REPLICAS_SETTING.exists(indexSettingsBuilder) == false) {
+        if (INDEX_NUMBER_OF_REPLICAS_SETTING.exists(indexSettingsBuilder) == false
+            || indexSettingsBuilder.get(SETTING_NUMBER_OF_REPLICAS) == null) {
```

Comment:
Based on the issue you shared, there is also another way in which you can get the intended behaviour. The below code that result in NPE can be revised to the following change -
https://github.com/opensearch-project/OpenSearch/blob/e288962d2870efa28b3dc67f5bcb7ad1a2f9d57f/server/src/main/java/org/opensearch/index/IndexSettings.java#L1262-L1264

```
    public int getNumberOfReplicas() {
        return IndexMetadata.INDEX_NUMBER_OF_REPLICAS_SETTING.get(settings);
    }
```

Can we also evaluate this option?

Reply:
Suggested code will get default value from [here](https://github.com/opensearch-project/OpenSearch/blob/9a876245f01f15cf58d3587f6b651e8181658a2b/server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java#L239) which is always 1 regardless of cluster setting.